sqlglot/dialects/snowflake.py

```diff
@@ -236,6 +237,19 @@ class Snowflake(Dialect):
         "ff6": "%f",
     }
 
+    @classmethod
+    def quote_identifier(cls, expression: E, identify: bool = True) -> E:
+        # This disables quoting DUAL in SELECT ... FROM DUAL, because Snowflake treats an
+        # unquoted DUAL keyword in a special way and does not map it to a user-defined table
+        if (
+            isinstance(expression, exp.Identifier)
+            and isinstance(expression.parent, exp.Table)
+            and expression.name.lower() == "dual"
+        ):
+            return t.cast(E, expression)
+
+        return Dialect.quote_identifier(expression, identify=identify)
```

Comment:
This is problematic because it's changing the receiver class and hence we don't get the `quote_identifier` behavior for Snowflake.. Trying to patch it now.